website/src/components/subscriptions/backendApi/backendService.ts

```diff
@@ -0,0 +1,102 @@
+import axios, { type AxiosInstance, type AxiosResponse } from 'axios';
+import { z, ZodSchema } from 'zod';
+import {
+    type SubscriptionPutRequest,
+    type SubscriptionRequest,
+    subscriptionResponseSchema,
+    triggerEvaluationResponseSchema,
+} from '../../../types/Subscription.ts';
+
+class ApiService {
+    private axiosInstance: AxiosInstance;
+
+    constructor(baseURL: string) {
+        this.axiosInstance = axios.create({ baseURL });
+    }
+
+    public async get<Response>(url: string, schema: ZodSchema<Response>): Promise<Response> {
```

Comment:
As discussed: we use useQuery to fetch from the server. This will produce an error, when a fetch failed. So we dont need extensive error handling here. It would be nice to have, but we defer this to a later point, when we use the backendService differently.